$(FSMONITOR) starts with a hyphen, so it's interpreted as an option.
$(NAME) is empty, so -blob is interpreted as an option.

--- a/fsmonitor/linux/Makefile
+++ b/fsmonitor/linux/Makefile
@@ -26,4 +26,4 @@
 
 clean::
 	rm -f $(DIR)/*.cm[iox] $(DIR)/*.o $(DIR)/*~
-	rm -f $(FSMONITOR)$(EXEC_EXT)
\ No newline at end of file
+	rm -f -- $(FSMONITOR)$(EXEC_EXT)
--- a/Makefile
+++ b/Makefile
@@ -354,8 +354,8 @@
 clean::
 	-$(RM) *.log *.aux *.log *.dvi *.out *.bak
 	-$(RM) -r obsolete
-	-$(RM) $(NAME) $(NAME).exe
-	-$(RM) $(NAME)-blob.o
+	-$(RM) -- $(NAME) $(NAME).exe
+	-$(RM) -- $(NAME)-blob.o
 
 clean::
 	$(MAKE) -C ubase clean
